mycore-base/src/main/java/org/mycore/access/mcrimpl/MCRAccessData.java

```diff
@@ -25,18 +25,29 @@
 
 import java.util.Date;
 
+import org.mycore.common.MCRUserInformation;
+
 public class MCRAccessData {
     private String userID;
+    
+    private MCRUserInformation userInformation;
 
     private Date date;
 
     private MCRIPAddress ip;
 
+    @Deprecated
```

Comment:
I would delete the old constructor, because it is "only" protected. You can still compare the UserID in the information with a passed UserID in UserInformation.

Reply:
MCRAcessRule uses this constructor. It should stay there until cleanup after next release.